### PR TITLE
docs(javascript engine): add engine names explicitly

### PR DESCRIPTION
**Pull request by ponchique**
### Description
In the PR, the engine names (with links) were explicitly added to the main document.

### Motivation

It was done to provide a solid starting point for engine studies and offer a quick overview for users interested in high-level information.

### CODE REVIEW

File: files/en-us/glossary/engine/javascript/index.md [hunk 1]
```diff
@@ -12,6 +12,12 @@ In a browser, the JavaScript engine operates together with the rendering engine
 
 Do not confuse JavaScript engines with {{glossary("engine/rendering", "rendering engines")}}, which are also crucial parts of browsers.
 
+Common JavaScript engines include:
+
+- [V8](https://v8.dev/)
+- [SpiderMonkey](https://spidermonkey.dev/)
+- [WebKit](https://webkit.org/)
```
Comment (hamishwillee) [inline files/en-us/glossary/engine/javascript/index.md:19]: See line above - this isn't a javascript engine.

```suggestion
```

**Review by hamishwillee — APPROVED** [review]
Thanks @ponchique . 

@Josh-Cena This looks reasonable to me, but deferring to you in case there is some reason not to include a list.

**Comment by Josh-Cena** [comment]
The Safari JS is called JavaScriptCore. Could you link to that too? We should definitely also link to https://developer.mozilla.org/en-US/docs/Web/JavaScript/JavaScript_technologies_overview#javascript_implementations for all the non-browser engines.

File: files/en-us/glossary/engine/javascript/index.md (hunk 1 above)
Comment (ponchique) [reply files/en-us/glossary/engine/javascript/index.md]: @hamishwillee thanks for pointing this out, put the wrong link. will fix it

File: files/en-us/glossary/engine/javascript/index.md [hunk 2]
```diff
@@ -12,9 +12,16 @@ In a browser, the JavaScript engine operates together with the rendering engine
...
+Common JavaScript engines include:
+
+- [V8](https://v8.dev/)
+- [SpiderMonkey](https://spidermonkey.dev/)
+- [JavaScriptCore](https://developer.apple.com/documentation/javascriptcore)
+
 ## See also
 
 - [JavaScript engine](https://en.wikipedia.org/wiki/JavaScript_engine) on Wikipedia
+- [Browser-Based vs. Non-Browser Engines](/en-US/docs/Web/JavaScript/JavaScript_technologies_overview#javascript_implementations)
```
Comment (Josh-Cena) [inline files/en-us/glossary/engine/javascript/index.md:24]:
```suggestion
- [JavaScript implementations](/en-US/docs/Web/JavaScript/JavaScript_technologies_overview#javascript_implementations)
```

**Review by Josh-Cena — APPROVED** [review]

**Pull request merged**